### PR TITLE
Improve Error Handling for tt-steps (#138)

### LINKED ISSUE #138: Improve Error Handling for tt-steps
### Describe the feature request

As a plugin maintainer I want to increase the usability of the plugin by meaningful error messages for standard issues.

### Upstream changes

* [x] https://github.com/jenkinsci/ecu-test-execution-plugin/blob/main/src/main/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/steps

### Acceptance criteria

* [x] no Stack Traces for known errors
* [x] add hints how to solve issues in error log

### PR DESCRIPTION
**Pull request by MarioFuchsTT**
Resolves #138

### CODE REVIEW

File: src/main/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/steps/RunTestFolderStep.groovy [hunk 1]
```diff
@@ -166,30 +167,36 @@ class RunTestFolderStep extends RunTestStep {
...
         if (scanMode == ScanMode.PROJECTS_ONLY) {
             return pkgFiles
         }
 
-        final TestPackageScanner scanner = new TestPackageScanner(testFolder, isRecursive, context)
-        pkgFiles = scanner.scanTestFiles()
+        try {
+            final TestPackageScanner scanner = new TestPackageScanner(testFolder, isRecursive, context)
+            pkgFiles = scanner.scanTestFiles()
+        } catch (IOException | InterruptedException e) {
```
Comment (MartinGroscheTT) [inline src/main/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/steps/RunTestFolderStep.groovy:195]: Do not found InterruptedException

File: src/main/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/steps/StartToolStep.groovy [hunk 2]
```diff
@@ -207,7 +207,9 @@ class StartToolStep extends Step {
                     if (ProcessUtil.killTTProcesses(timeout)) {
                         listener.logger.println("Stopped tracetronic tools successfully.")
                     } else {
-                        throw new TimeoutException("Timeout of ${this.timeout} seconds exceeded for stopping tracetronic tools!")
+                        throw new AbortException(
+                                "Timeout of ${this.timeout} seconds exceeded for stopping tracetronic tools!" +
+                                "Please ensure that tracetronic tools are not already stopped.")
```
Comment (MartinGroscheTT) [inline src/main/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/steps/StartToolStep.groovy:212]: It is also possible that some process is blocked and we run into a timeout as a result

File: src/main/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/steps/StartToolStep.groovy [hunk 3]
```diff
@@ -296,7 +301,9 @@ class StartToolStep extends Step {
             try {
                  RestApiClientFactory.getRestApiClient(envVars.get('ET_API_HOSTNAME'), envVars.get('ET_API_PORT'), timeout)
             } catch (ApiException e) {
-                throw new TimeoutException("Timeout of ${this.timeout} seconds exceeded for connecting to ${toolName}!")
+                throw new AbortException(
+                        "Timeout of ${this.timeout} seconds exceeded for connecting to ${toolName}!" +
+                        "Please ensure the tool is running and its API endpoint is accessible.")
```
Comment (MartinGroscheTT) [inline src/main/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/steps/StartToolStep.groovy:306]: If the tool is started correctly, the endpoint can be reached. The user has no possibility to check the endpoint separately

**Review by MartinGroscheTT — CHANGES_REQUESTED** [review]
* do we need to adapt this [line ](https://github.com/jenkinsci/ecu-test-execution-plugin/blob/3058dad11ed769d8bd6e82ca5dee2a02c291cb6d/src/main/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/steps/StartToolStep.groovy#L151) as well
* getExeFileOnNode still throws an IllegalArgumentException -> please check all classes for adaption needed
* no tests found for those code changes

**Comment by MarioFuchsTT** [comment]
> * do we need to adapt this [line ](https://github.com/jenkinsci/ecu-test-execution-plugin/blob/3058dad11ed769d8bd6e82ca5dee2a02c291cb6d/src/main/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/steps/StartToolStep.groovy#L151) as well

For those, exceptions are now caught and converted to an AbortException, along with their messages.


> * getExeFileOnNode still throws an IllegalArgumentException -> please check all classes for adaption needed

This will now be handled within the catch block of its calling method.

**Review by MartinGroscheTT — APPROVED** [review]

**Pull request merged**